src/datepicker/datepicker.js

```diff
@@ -580,24 +595,25 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
         }
       }
 
-      ngModel.$validators.date = validator;
-      ngModel.$parsers.unshift(parseDate);
-
-      ngModel.$formatters.push(function (value) {
-        scope.date = value;
-         return ngModel.$isEmpty(value) ? value : dateFilter(value, dateFormat);
-      });
+      if (!isHtml5DateInput) {
+        // Internal API to maintain the correct ng-invalid-[key] class
+        ngModel.$$parserName = 'date';
+        ngModel.$validators.date = validator;
+        ngModel.$parsers.unshift(parseDate);
+        ngModel.$formatters.push(function (value) {
+          scope.date = value;
+           return ngModel.$isEmpty(value) ? value : dateFilter(value, dateFormat);
+        });
+      }
 
       // Inner change
       scope.dateSelection = function(dt) {
         if (angular.isDefined(dt)) {
           scope.date = dt;
         }
-        if (dateFormat) {
```

Comment:
Hm.. Good point, the directive should throw immediately if dateFormat is falsey. It's undefined behavior and definitely an error in the use of this directive.